src/Makefile.crc32c.include

```diff
@@ -33,41 +33,49 @@ else
 CRC32C_CPPFLAGS_INT += -DBYTE_ORDER_BIG_ENDIAN=0
 endif
 
-crc32c_libcrc32c_a_CPPFLAGS = $(AM_CPPFLAGS) $(CRC32C_CPPFLAGS_INT) $(CRC32C_CPPFLAGS)
-crc32c_libcrc32c_a_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
-
-crc32c_libcrc32c_a_SOURCES =
-crc32c_libcrc32c_a_SOURCES += crc32c/include/crc32c/crc32c.h
-crc32c_libcrc32c_a_SOURCES += crc32c/src/crc32c_arm64.h
-crc32c_libcrc32c_a_SOURCES += crc32c/src/crc32c_arm64_check.h
-crc32c_libcrc32c_a_SOURCES += crc32c/src/crc32c_internal.h
-crc32c_libcrc32c_a_SOURCES += crc32c/src/crc32c_prefetch.h
-crc32c_libcrc32c_a_SOURCES += crc32c/src/crc32c_read_le.h
-crc32c_libcrc32c_a_SOURCES += crc32c/src/crc32c_round_up.h
-crc32c_libcrc32c_a_SOURCES += crc32c/src/crc32c_sse42_check.h
-crc32c_libcrc32c_a_SOURCES += crc32c/src/crc32c_sse42.h
-
-crc32c_libcrc32c_a_SOURCES += crc32c/src/crc32c.cc
-crc32c_libcrc32c_a_SOURCES += crc32c/src/crc32c_portable.cc
+crc32c_libcrc32c_la_CPPFLAGS = $(AM_CPPFLAGS) $(CRC32C_CPPFLAGS_INT) $(CRC32C_CPPFLAGS)
+
+# Specify -static in both CXXFLAGS and LDFLAGS so libtool will only build a
+# static version of this library. We don't need a dynamic version, and a dynamic
+# version can't be used on windows anyway because the library doesn't currently
+# export DLL symbols.
+crc32c_libcrc32c_la_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS) -static
```

Comment:
Heh, yes, that's about it. libtool manages to defy intuition at about every turn, but at least the outcome here is reasonable (I hope!).

By default for every lib it wants to build a shared and static version, and because of the differing export semantics it wants to compile one set of objects to go into the shared library, and another set for the static library.

Adding `-static` in LDFLAGS tells it to disable any linking, but it'll still build the "shared" objects as well.
Adding `-static` in CXXFLAGS tells it that there will be nothing to link in the end, so there's no point in building the "shared" objects.

As for argument parsing, yes, it's really nasty. libtool gobbles up parameters passed to the compiler/linker and attempts to substitute the correct actions. It gets hairy when there's ambiguity with keywords (like "-static").

It's a dinosaur :(

Reply:
> Adding `-static` in CXXFLAGS tells it that there will be nothing to link in the end, so there's no point in building the "shared" objects.

Hmm, I'm not sure what's going on exactly, but _removing_ `-static` from `leveldb_libleveldb_la_CXXFLAGS`
```diff
--- a/src/Makefile.leveldb.include
+++ b/src/Makefile.leveldb.include
@@ -43,7 +43,7 @@ leveldb_libleveldb_la_CPPFLAGS = $(AM_CPPFLAGS) $(LEVELDB_CPPFLAGS_INT) $(LEVELD
 # static version of this library. We don't need a dynamic version, and a dynamic
 # version can't be used on windows anyway because the library doesn't currently
 # export DLL symbols.
-leveldb_libleveldb_la_CXXFLAGS = $(filter-out -Wconditional-uninitialized -Werror=conditional-uninitialized -Wsuggest-override -Werror=suggest-override, $(AM_CXXFLAGS)) $(PIE_FLAGS) -static
+leveldb_libleveldb_la_CXXFLAGS = $(filter-out -Wconditional-uninitialized -Werror=conditional-uninitialized -Wsuggest-override -Werror=suggest-override, $(AM_CXXFLAGS)) $(PIE_FLAGS)
 leveldb_libleveldb_la_LDFLAGS = $(AM_LDFLAGS) -static
 
 leveldb_libleveldb_la_SOURCES=
```

changes nothing in building `libleveldb.la` -- I've compared verbose build logs and build directories.

But many others `-libtool: compile:` calls are _not_ invoked twice, and `.libs/libbitcoinconsensus.so.0.0.0` is not created.

---

I used the following configuration:
```
$ ./configure --enable-suppress-external-warnings --disable-wallet --without-gui --without-utils --disable-tests --disable-bench --disable-fuzz-binary --without-zmq CC=clang CXX=clang++
```